proxy/reverse.go

```diff
@@ -118,6 +124,9 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		redirectRequest(proxyreq, ep.URL)
 
 		res, err = p.transport.RoundTrip(proxyreq)
+		if atomic.LoadInt32(&requestCanceled) == 1 {
```

Comment:
is it more appropriate to name it 'closed'? We stop retrying because the client side is closed.